Parliament.Data.Api.FixedQuery/Controllers/HouseController.cs

```diff
@@ -177,77 +177,100 @@ public Graph CurrentMembers(string id)
-        :houseIncumbencyHasHouse ?house .
-    ?seat
+        a :HouseIncumbency .
+
+    ?houseSeat
         a :HouseSeat ;
+        :houseSeatHasSeatIncumbency ?incumbency ;
         :houseSeatHasConstituencyGroup ?constituency .
+
+    ?incumbency
+        a :Incumbency ;
+        :incumbencyHasMember ?person ;
+        :incumbencyStateDate ?startDate .
+
+    ?constituency
+        a :ConstituencyGroup ;
+        :constituencyGroupName ?constituencyName .
+
+    ?person
+        a :Person ;
+        :personGivenName ?givenName ;
+        :personFamilyName ?familyName ;
+        <http://example.com/F31CBD81AD8343898B49DC65743F0BDF> ?displayAs ;
+        <http://example.com/A5EE13ABE03C4D3A8F1A274F57097B6C> ?listAs ;
+        :partyMemberHasPartyMembership ?partyMembership .
+
     ?partyMembership
         a :PartyMembership ;
+        :partyMembershipStartDate ?membershipStartDate ;
+        :partyMembershipEndDate ?membershipEndDate ;
         :partyMembershipHasParty ?party .
+
     ?party
         a :Party ;
         :partyName ?partyName .
-    ?constituency
-        a :ConstituencyGroup ;
-        :constituencyGroupName ?constituencyName .
+
 }
 WHERE {
     BIND(@houseid AS ?house)
   	?house 
         a :House ;
         :houseName ?houseName .
-    ?person 
-        a :Member ;
-        :partyMemberHasPartyMembership ?partyMembership .
-    FILTER NOT EXISTS { ?partyMembership a :PastPartyMembership . }
-    ?partyMembership :partyMembershipHasParty ?party .
-    ?party :partyName ?partyName .
-    ?incumbency :incumbencyHasMember ?person ;
-        :incumbencyStartDate ?incumbencyStartDate .
-    FILTER NOT EXISTS { ?incumbency a :PastIncumbency . }
+ OPTIONAL {
     {
-        ?incumbency :houseIncumbencyHasHouse ?house .
-        BIND(?incumbency AS ?houseIncumbency)
-	}
-    UNION {
-        ?incumbency :seatIncumbencyHasHouseSeat ?seat .
-        ?seat 
-        	:houseSeatHasHouse ?house ;
-        	:houseSeatHasConstituencyGroup ?constituency .
-        ?constituency :constituencyGroupName ?constituencyName .
-        BIND(?incumbency AS ?seatIncumbency)
-	}
-    OPTIONAL { ?person :personGivenName ?givenName . }
-    OPTIONAL { ?person :personFamilyName ?familyName . }
-    OPTIONAL { ?person <http://example.com/F31CBD81AD8343898B49DC65743F0BDF> ?displayAs } .
+    ?house :houseHasHouseSeat ?houseSeat .
+    ?houseSeat :houseSeatHasConstituencyGroup ?constituency .
+    ?constituency
+         a :ConstituencyGroup ;
+         :constituencyGroupName ?constituencyName .
+    ?houseSeat :houseSeatHasSeatIncumbency ?incumbency .
+    ?incumbency :incumbencyHasMember ?person .
+    ?incumbency
+         a :Incumbency ;
+         :incumbencyStartDate ?startDate .
+    }
+    UNION
+    {
+    ?house :houseHasHouseIncumbency ?houseIncumbency .
+    ?houseIncumbency
+         a :Incumbency ;
+         :incumbencyStartDate ?stateDate .
+    ?houseIncumbency :incumbencyHasMember ?person .
+    BIND(?houseIncumbency AS ?incumbency)
```

Comment:
I'm wondering if you bind ?houseIncumbency as ?incumbency here, should we not do the same for the ?seatIncumbency?